### ISSUE TITLE
Support `loss.backward(create_graph=True)` in the Trainer

### ISSUE THREAD

**Issue by hanlint** [mosaicml/composer#6]
Methods such as `AdaHessian` or similar need support for this feature.
